cmd/kubeadm/app/cmd/phases/reset/unmount_linux.go

```diff
@@ -45,8 +45,10 @@ func unmountKubeletDirectory(absoluteKubeletRunDirectory string) error {
 		if len(m) < 2 || !strings.HasPrefix(m[1], absoluteKubeletRunDirectory) {
 			continue
 		}
-		if err := syscall.Unmount(m[1], 0); err != nil {
+		if err := syscall.Unmount(m[1], syscall.MNT_DETACH); err != nil {
 			klog.Warningf("[reset] Failed to unmount mounted directory in %s: %s", absoluteKubeletRunDirectory, m[1])
```

Comment:
With this change, `KubeletRunDirectory` will not be cleaned if an error is returned here.

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/phases/reset/unmount_linux.go#L42-L51
If we return err here, the loop will end here. However, I think we probably want to try to unmount others as well.(We should take care this as well.)

BTW, +1 for @SataQiu's comments https://github.com/kubernetes/kubeadm/issues/2923.
> So unmountKubeletDirectory is just best-effort, and users performing kubeadm reset usually means that they want to clean up the existing data on the node, right?
> If users want to keep the data, I think they need to make a backup before executing kubeadm reset.

Reply:
Okay, I'll gather errs and return outside the loop~